pkg/visor/visor.go

```diff
@@ -501,7 +554,156 @@ func (node *Node) SetAutoStart(appName string, autoStart bool) error {
 
 	appConf.AutoStart = autoStart
 	node.appsConf[appName] = appConf
-	return nil
+
+	return node.updateConfigAppAutoStart(appName, autoStart)
+}
+
+func (node *Node) updateConfigAppAutoStart(appName string, autoStart bool) error {
+	if node.confPath == nil {
+		return nil
+	}
+
+	config, err := node.readConfig()
+	if err != nil {
+		return err
+	}
+
+	node.logger.Infof("Saving auto start = %v for app %v to config", autoStart, appName)
+
+	changed := false
+
+	for i := range config.Apps {
+		if config.Apps[i].App == appName {
+			config.Apps[i].AutoStart = autoStart
+			changed = true
+			break
+		}
+	}
+
+	if !changed {
+		return nil
+	}
+
+	return node.writeConfig(config)
+}
+
+// SetSocksPassword sets skysocks password.
+func (node *Node) SetSocksPassword(password string) error {
```

Comment:
Is this one has to be exported?

Reply:
@Darkren You're right, it doesn't have to. I just implemented it in the same way as `SetAutoStart`. Fixed, thanks.